nucypher/blockchain/eth/sol/source/contracts/WorkLock.sol

```diff
@@ -0,0 +1,147 @@
+pragma solidity ^0.5.3;
+
+
+import "zeppelin/math/SafeMath.sol";
+import "contracts/NuCypherToken.sol";
+import "contracts/MinersEscrow.sol";
+
+
+/**
+* @notice The WorkLock distribution contract
+**/
+contract WorkLock {
+    using SafeMath for uint256;
+
+    event Bid(address indexed miner, uint256 depositedETH, uint256 claimedTokens);
+    event Claimed(address indexed miner, uint256 claimedTokens);
+    event Refund(address indexed miner, uint256 refundETH, uint256 workDone);
+
+    struct WorkInfo {
+        uint256 depositedETH;
+        uint256 workDone;
+        bool claimed;
+    }
+
+    NuCypherToken public token;
+    MinersEscrow public escrow;
+    uint256 public startBidDate;
+    uint256 public endBidDate;
+    // ETH -> NU
+    uint256 public depositRate;
+    // Work (reward in NU) -> ETH
+    uint256 public refundRate;
```

Comment:
However we end up calculating `refundETH`, it may be prudent to hold off specifying associated parameters (like `refundRate`) until we've seen work activity on the network for some months. Otherwise we're shooting blindly, and it would justifiably upset stakers if we were forced to adjust it after the fact – it actually could be better to leave a variable deliberately TBD than to change it later (unless this change emerges via a community-driven process). 

Of course this is part of a wider debate about governance and the role of the NuCypher team following the launch of the network. There is (rightly) skepticism about networks that remain almost entirely centralized with vague promises of foundation control loosening later. At the same time, you can make the argument that in at least the first months after launch, there will be some dependence on the NuCypher team regardless (e.g. efforts to drive network adoption). Furthermore, stakers may prefer the reassurance that some key parameters can be adjusted to avoid network shocks or capitulation. 

@mswilkison @michwill

Reply:
IMO, this parameter mainly depends on the inflation rate. E.g. if this refund rate is the same as deposit rate, you'r have ~1 year to get fully refunded (depends on how you stake though) if you have 100% initial emission rate.
If this is too aggressive and we make it lower, that, imo, should affect the refund rate.

How about we make the refund rate linked to emission rate, so that the full refund time is 1 year, for example?

Otherwise, if we left it TBD, I think participants would be very meh on it